### PR TITLE
Feature/311 confirmation dialog

### PR DESCRIPTION
**Pull request by Muhaddatha**
- Confirm user actions (importing file, tutorial) when a model has already been simulated or instantiated.

### CODE REVIEW

**Review by ddelpiano — APPROVED** [review]
Tested and works fine, code and implementation review left to @enicolasgomez and @filippomc since it was discussed with them.

**Comment by ddelpiano** [comment]
@Muhaddatha could you please solve the conflicts ? Thx

**Review by ddelpiano — APPROVED** [review]

File: webapp/components/topbar/Topbar.js [hunk 1]
```diff
@@ -85,6 +86,27 @@ class Topbar extends Component {
         }
         break;
       }
+      case 'handleTutorial': {
+        const [action, payload] = click.parameters;
+        if (this.props.modelState === MODEL_STATE.INSTANTIATED || this.props.modelState === MODEL_STATE.SIMULATED) {
+          this.props.openConfirmationDialog({
+            title: 'Warning',
+            message: 'A NetPyNE model has already been instantiated or simulated.'
```
Comment (enicolasgomez) [inline webapp/components/topbar/Topbar.js:94]: Let's put this message into a global variable DEFAULT_CONFIRMATION_DIALOG_MESSAGE as it been re-used.

File: webapp/components/topbar/dialogs/ImportExportHLS.js [hunk 2]
```diff
@@ -179,8 +180,21 @@ class ImportExportHLS extends React.Component {
     }
   }
 
+  handleConfirmation (command, args) {
+    this.props.openConfirmationDialog({
+      title: 'Warning',
+      message: 'A NetPyNE model has already been instantiated or simulated.'
```
Comment (enicolasgomez) [inline webapp/components/topbar/dialogs/ImportExportHLS.js:186]: Use DEFAULT_CONFIRMATION_DIALOG_MESSAGE here too

**Comment by enicolasgomez** [comment]
@Muhaddatha @ddelpiano looking good! one minor change request and we will be good to go

**Review by enicolasgomez — APPROVED** [review]

File: webapp/components/topbar/dialogs/ActionDialog.js [hunk 3]
```diff
@@ -38,8 +38,12 @@ class ActionDialog extends React.Component {
   performAction = () => {
     if (this.props.command) {
       if (this.props.isFormValid === undefined || this.props.isFormValid()) {
-        GEPPETTO.trigger(GEPPETTO.Events.Show_spinner, this.props.message);
-        this.props.pythonCall(this.props.command, this.props.args);
+        if (typeof this.props.callback !== 'undefined' || this.props.callback !== null) {
+          this.props.callback(this.props.command, this.props.args);
+        } else {
+          GEPPETTO.trigger(GEPPETTO.Events.Show_spinner, this.props.message);
```
Comment (filippomc) [inline webapp/components/topbar/dialogs/ActionDialog.js:44]: Legacy spinner trigger: the limitation of this approach is that parallel asynchronous operations may trigger the spinner and the first closer closes for all. I suggest to always use The specific action https://github.com/MetaCell/geppetto-meta/blob/master/geppetto.js/geppetto-client/src/common/actions/actions.js#L141 and specify a close action (in this case would be a receive python message I suppose). cc @ddelpiano know you were on it, what do you suggest here?

Comment (enicolasgomez) [reply webapp/components/topbar/dialogs/ActionDialog.js:44]: @Muhaddatha just for reference, this has just been merged to development:

https://github.com/MetaCell/NetPyNE-UI/pull/466

**Pull request merged**